### PR TITLE
Remove necessity to have a separate worker script

### PR DESCRIPTION
**Pull request by fergalhanley**
I removed the necessity to load a separate script to for the Worker object. Instead I convert the worker function to a string use it to create a Blob where I can get a local URL reference to and pass to the Worker constructor. 

I'm getting the same score results so it appears to have no performance impact on Fireball running, and it shouldn't, it's the same script running.

Additionally I added a build step to use bowrap so the index.min.js is loadable directly to the browser and Fireball is available globally. Also registered fireball with bower. It's using this fork as the repo but will move over to main branch if accepted.

Peace.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,23 +5,34 @@ Fireball is a small script that runs when your web page is loaded. It generates
...
 
+### Running fireball, the simple way
 ```javascript
-server.use('/fireball-js', express.static(path.resolve(__dirname, '../../node_modules/fireball-js')));
+var Fireball = require('fireball-js');
+Fireball.run();
 ```
 
-### Running fireball, the simple way
+or if adding the script directly or using bower the Fireball will be already available:
```
Comment (davidgilbertson) [inline README.md]: "...the Fireball will be..."

Maybe just "Fireball" or "the Fireball object".

File: src/index.js [hunk 2]
```diff
@@ -118,7 +118,30 @@ function appendClasses(options) {
     }
 }
 
-function runFireball(options) {
+const fireballWorker = function(){
+    'use strict';
```
Comment (davidgilbertson) [inline src/index.js:122]: ES6 doesn't need `'use strict';`

File: src/index.js [hunk 3]
```diff
@@ -131,7 +154,14 @@ function runFireball(options) {
     const debug = options.debug || false;
     const runs = options.runs || 7;
     let count = 0;
-    const fbWorker = new Worker('/fireball-js/fireball_worker.js');
+    var blobURL = URL.createObjectURL(new Blob(
```
Comment (davidgilbertson) [inline src/index.js]: `var` can be `const`

File: src/index.js [hunk 4]
```diff
@@ -131,7 +154,14 @@ function runFireball(options) {
...
+        ['(', fireballWorker.toString(), ')()'],
```
Comment (davidgilbertson) [inline src/index.js]: Minor: could be template string

``` js
[`(${fireballWorker.toString()})()`]
```

File: src/index.js [hunk 5]
```diff
@@ -131,7 +154,14 @@ function runFireball(options) {
...
+        {type: 'application/javascript'}
+    ));
+
+    const fbWorker = new Worker(blobURL);
+
+    URL.revokeObjectURL(blobURL);
```
Comment (davidgilbertson) [inline src/index.js:164]: This blob stuff is very clever!

File: src/index.js [hunk 6]
```diff
@@ -180,7 +210,7 @@ function addCallback(callback) {
 }
 
 export default {
-    run: runFireball,
+    run: run,
     getScore: getScore,
-    onSuccess: addCallback
+    onSuccess: onSuccess
 };
```
Comment (davidgilbertson) [inline src/index.js:216]: Minor: could be:

``` js
export default {
    run,
    getScore,
    onSuccess
};
```

**Comment by davidgilbertson** [comment]
Dude this is great, makes it so much easier to use. My only comments are minor, if you don't feel like making changes just say the word and I'll click merge.

File: src/index.js (hunk 2 above)
Comment (fergalhanley) [inline src/index.js:122]: Yep but we need 'use strict' to be added to the function for the worker script.

File: src/index.js (hunk 5 above)
Comment (fergalhanley) [inline src/index.js:164]: Yea used it for my beatworker so had the code ready to go. Since I'm using fireball it made sense to update it.

**Comment by fergalhanley** [comment]
No worries. Feedback in :)

**Comment by davidgilbertson** [comment]
Merged, thanks heaps!

**Pull request merged**